platform/ios/app/MBXViewController.m

```diff
@@ -223,38 +223,39 @@ @implementation MBXViewController
 
 #pragma mark - Setup & Teardown
 
-+ (void)initialize
-{
-    if (self == [MBXViewController class])
-    {
-        [[NSUserDefaults standardUserDefaults] registerDefaults:@{
-            @"MBXUserTrackingMode": @(MGLUserTrackingModeNone),
-            @"MBXShowsUserLocation": @NO,
-            @"MBXDebug": @NO,
-        }];
-    }
-}
-
 - (void)viewDidLoad
 {
     [super viewDidLoad];
 
-    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(saveState:) name:UIApplicationDidEnterBackgroundNotification object:nil];
-    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(restoreState:) name:UIApplicationWillEnterForegroundNotification object:nil];
-    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(saveState:) name:UIApplicationWillTerminateNotification object:nil];
+    // Keep track of current map state and debug preferences,
+    // saving and restoring when the below notifications are posted
+    self.currentState =  [MBXStateManager sharedManager].currentState;
 
-    [self restoreState:nil];
+    if (self.currentState != nil) {
+        [self restoreMapState:nil];
+    }
+
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(saveCurrentMapState:) name:UIApplicationDidEnterBackgroundNotification object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(restoreMapState:) name:UIApplicationWillEnterForegroundNotification object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(saveCurrentMapState:) name:UIApplicationWillTerminateNotification object:nil];
 
     if ([MGLAccountManager accessToken].length)
     {
         self.styleIndex = -1;
         [self cycleStyles:self];
     }
 
-    self.debugLoggingEnabled = [[NSUserDefaults standardUserDefaults] boolForKey:@"MGLMapboxMetricsDebugLoggingEnabled"];
-    self.mapView.showsScale = YES;
-    self.mapView.showsUserHeadingIndicator = YES;
-    self.mapView.experimental_enableFrameRateMeasurement = YES;
+    self.currentState.debugLoggingEnabled = YES;
```

Comment:
This will always override the state, add this code from line 248-257  as part of an else in line 236